client/network/src/config.rs

```diff
@@ -103,6 +107,9 @@ pub struct Params<B: BlockT, H: ExHashT> {
 
 	/// Registry for recording prometheus metrics to.
 	pub metrics_registry: Option<Registry>,
+
+	/// Whether a node is in a permissioned network or not.
+	pub permissioned_network: bool,
```

Comment:
Can we not determine that dynamically, based on whether `well_known_keys::NODE_ALLOWLIST` exists instead?